lint/clippy.bzl

```diff
@@ -0,0 +1,191 @@
+        command = """
+            if [ -s $2 ]; then
+                echo '1' > $1
+            else
+                echo '0' > $1
+            fi
+        """,
+    )
+
+# buildifier: disable=function-docstring
+def _clippy_aspect_impl(target, ctx):
+    if not should_visit(ctx.rule, ctx.attr._rule_kinds):
+        return []
+
+    clippy_bin = ctx.toolchains[Label("@rules_rust//rust:toolchain_type")].clippy_driver
+
+    files_to_lint = filter_srcs(ctx.rule)
+    if ctx.attr._options[LintOptionsInfo].fix:
+        print("WARNING: `fix` is not supported yet for clippy. Please follow https://github.com/aspect-build/rules_lint/issues/385 for updates.")
+
+    outputs, info = output_files(_MNEMONIC, target, ctx)
+
+    if len(files_to_lint) == 0:
+        noop_lint_action(ctx, outputs)
+        return [info]
+
+    crate_info = rust_clippy_action.get_clippy_ready_crate_info(target, ctx)
+    if not crate_info:
+        noop_lint_action(ctx, outputs)
+        return [info]
+
+    extra_options = []
+    # FIXME: Implement support for --fix mode. Clippy has a --fix flag, but our patcher doesn't currently support running an action through a macro.
+    #        We have to either
+    #           (1) modify the patcher so that it can run an action through a macro, or
+    #           (2) modify rules_rust so that it gives us a struct with a command line we can run it with the patcher.
+
+    human_success_indicator = ctx.actions.declare_file(OUTFILE_FORMAT.format(label = target.label.name, mnemonic = _MNEMONIC, suffix = "human_success_indicator"))
+    rust_clippy_action.action(
+        ctx,
+        clippy_executable = clippy_bin,
+        process_wrapper = ctx.executable._process_wrapper,
+        crate_info = crate_info,
+        config = ctx.file._config_file,
+        output = outputs.human.out,
+        success_marker = human_success_indicator,
+        cap_at_warnings = True,  # We don't want to crash the process if there are clippy errors, we just want to report them.
+        extra_clippy_flags = extra_options,
+    )
+    _marker_to_exit_code(ctx, human_success_indicator, outputs.human.out, outputs.human.exit_code)
+
+    machine_success_indicator = ctx.actions.declare_file(OUTFILE_FORMAT.format(label = target.label.name, mnemonic = _MNEMONIC, suffix = "machine_success_indicator"))
+    rust_clippy_action.action(
+        ctx,
+        clippy_executable = clippy_bin,
+        process_wrapper = ctx.executable._process_wrapper,
+        crate_info = crate_info,
+        config = ctx.file._config_file,
+        output = outputs.machine.out,
+        success_marker = machine_success_indicator,
+        cap_at_warnings = True,
+        extra_clippy_flags = extra_options,
+        error_format = "json",
+    )
+    _marker_to_exit_code(ctx, machine_success_indicator, outputs.machine.out, outputs.machine.exit_code)
+
+    # FIXME: Rustc only gives us JSON output, which we can't turn into SARIF yet.
```

Comment:
In theory https://github.com/reviewdog/errorformat/blob/master/fmts/rust.go should guide us to the answer